client/data/settings/__tests__/hooks.js

```diff
@@ -155,102 +145,148 @@ describe( 'Settings hooks tests', () => {
-		} );
-	} );
-
-	describe( 'usePaymentRequestButtonTheme()', () => {
-		test( 'returns the value of getSettings().payment_request_button_theme', () => {
-			selectors = {
-				getSettings: jest.fn( () => ( {
-					payment_request_button_theme: 'dark',
-				} ) ),
-			};
-
-			const { result } = renderHook( () =>
-				usePaymentRequestButtonTheme()
-			);
-			const [ value ] = result.current;
+	const generatedHookExpectations = {
+		useEnabledPaymentMethodIds: {
+			hook: useEnabledPaymentMethodIds,
+			storeKey: 'enabled_payment_method_ids',
+			testedValue: [ 'foo', 'bar' ],
+			fallbackValue: [],
+		},
+		usePaymentRequestEnabledSettings: {
+			hook: usePaymentRequestEnabledSettings,
+			storeKey: 'is_payment_request_enabled',
+			testedValue: true,
+			fallbackValue: false,
+		},
+		usePaymentRequestButtonSize: {
+			hook: usePaymentRequestButtonSize,
+			storeKey: 'payment_request_button_size',
+			testedValue: 'large',
+			fallbackValue: '',
+		},
+		usePaymentRequestButtonType: {
+			hook: usePaymentRequestButtonType,
+			storeKey: 'payment_request_button_type',
+			testedValue: '',
+			fallbackValue: '',
+		},
+		usePaymentRequestButtonTheme: {
+			hook: usePaymentRequestButtonTheme,
+			storeKey: 'payment_request_button_theme',
+			testedValue: 'dark',
+			fallbackValue: '',
+		},
+		useIsStripeEnabled: {
+			hook: useIsStripeEnabled,
+			storeKey: 'is_stripe_enabled',
+			testedValue: true,
+			fallbackValue: false,
+		},
+		useTestMode: {
+			hook: useTestMode,
+			storeKey: 'is_test_mode_enabled',
+			testedValue: true,
+			fallbackValue: false,
+		},
+		useSavedCards: {
+			hook: useSavedCards,
+			storeKey: 'is_saved_cards_enabled',
+			testedValue: true,
+			fallbackValue: false,
+		},
+		useManualCapture: {
+			hook: useManualCapture,
+			storeKey: 'is_manual_capture_enabled',
+			testedValue: true,
+			fallbackValue: false,
+		},
+		useSeparateCardForm: {
+			hook: useSeparateCardForm,
+			storeKey: 'is_separate_card_form_enabled',
+			testedValue: true,
+			fallbackValue: false,
+		},
+		useAccountStatementDescriptor: {
+			hook: useAccountStatementDescriptor,
+			storeKey: 'statement_descriptor',
+			testedValue: 'foo',
+			fallbackValue: '',
+		},
+		useIsShortAccountStatementEnabled: {
+			hook: useIsShortAccountStatementEnabled,
+			storeKey: 'is_short_statement_descriptor_enabled',
+			testedValue: true,
+			fallbackValue: false,
+		},
+		useShortAccountStatementDescriptor: {
+			hook: useShortAccountStatementDescriptor,
+			storeKey: 'short_statement_descriptor',
+			testedValue: 'bar',
+			fallbackValue: '',
+		},
+		useDebugLog: {
+			hook: useDebugLog,
+			storeKey: 'is_debug_log_enabled',
+			testedValue: true,
+			fallbackValue: false,
+		},
+		usePaymentRequestLocations: {
+			hook: usePaymentRequestLocations,
+			storeKey: 'payment_request_button_locations',
+			testedValue: [ 'checkout', 'cart' ],
+			fallbackValue: [],
+		},
+	};
+
+	describe.each( Object.entries( generatedHookExpectations ) )(
```

Comment:
This is a set of tests that will be ran for each hook from the above variable. Since all of the settings should behave the same way, it made sense to group them here.